packages/react-resizable-panels/src/PanelResizeHandleRegistry.ts

```diff
@@ -73,9 +73,48 @@ export function registerResizeHandle(
     if (count === 1) {
       ownerDocumentCounts.delete(ownerDocument);
     }
+
+    recalculateIntersectingHandlesAfterUnregister(element);
   };
 }
 
+function recalculateIntersectingHandlesAfterUnregister(
+  excludedElement: HTMLElement
+) {
+  intersectingHandles.splice(0); // Clear current intersecting handles
+
+  registeredResizeHandlers.forEach((data) => {
+    if (data.element !== excludedElement) {
+      const intersects = checkIfHandleIntersects(data);
+      if (intersects) {
+        intersectingHandles.push(data);
+      }
+    }
+  });
+
+  if (intersectingHandles.length > 0) {
+    updateCursor();
+  } else {
+    resetGlobalCursorStyle();
+  }
+}
+
+function checkIfHandleIntersects(data: ResizeHandlerData): boolean {
```

Comment:
This function is also unnecessary, as `recalculateIntersectingHandles` already has logic for this.

Reply:
> I remember why I didn't re-review this. This was kind of a turn off to me. In my experience, ChatGPT generates code that looks plausible but is often wrong in som way.

No worries, that's understandable and the reason I suspected.

I'd been looking over this code for a while without figuring out what exactly I needed to write, so I made use of ChatGPT to give me a starting point. I wanted to be forthright about it being AI-generated code for exactly the reasons you mention, but in this case I know it's generated something along the right lines as I've been able to verify its solution through unit testing and linking the local version of the library against my project.

You're right that there are some parts of the solution that didn't look right, even to me, such as the following:

```ts
if (intersectingHandles.length > 0) {
  updateCursor();
} else {
  resetGlobalCursorStyle();
}
```

Since we know that `resetGlobalCursorStyle` will be called by `updateCursor` if necessary anyway. However, I wanted to give ChatGPT's solution "as-is" to start with, since I don't understand the codebase as well as you and any amendments to its code on my part might be worse than the amendments you would have made. For example, in this case I would probably try to amend this code to be:

```ts
if (intersectingHandles.length == 0) {
  resetGlobalCursorStyle();
}
```

But I'm not 100% confident that's the best solution here.

What I'll try to do is incorporate the feedback you've given on ChatGPT's "as-is" solution to produce something simpler that looks right, still passes the unit test (now updated with your suggested), and still fixes the issue when linked against my project.

Reply:
> Given that, wouldn't the recalculateIntersectingHandles method work for updating the intersecting handles?

It was a while ago but I remember now - my original plan was to try exactly this, but `recalculateIntersectingHandles` expects an argument of the form:

```ts
{
  target: EventTarget | null;
  x: number;
  y: number;
}
```

Elsewhere, this function is invoked in response to an event e.g. `handlePointerDown`, `handlePointerMove`, etc. There was no event data to pass it from `unregisterResizeHandle` so I wasn't sure how I'd actually make use of it.

Based on your feedback, I've had another look at it and this time just tried passing a set of dummy event data. This passes the unit test and appears to work fine in my project, and limits the production code changes to just three additional lines.

I really appreciate your patience with all this, by the way.

Reply:
Hey no worries. And sorry for the slow response. Work was very busy this week and I'm just circling back to this.

The fake event question is a good one. I think we would actually need the mouse's most recent (current) position for that approach work. (It's likely that nothing would intersect with a fake 0,0 coordinate, so it would effectively always remove the cursor.) We'd really want it to recalculate and only remove the cursor if no remaining (still mounted) panels intersect with the mouse.

Maybe there's a better way to approach this. We keep the most recent set of intersecting handles in `intersectingHandles`. Maybe we should just remove the unmounting handle from that list, and then based on the remaining set– update the cursor. (This would handle the case where the set was empty and re removed the cursor entirely _as well as_ the case where it went from e.g. 2 handles with a "+" cursor to only one.)

Does this make sense? Sorry this only got half of my attention last week so I probably sent you down the wrong path.